Specs/Renderer/FramebufferManagerSpec.js

```diff
@@ -0,0 +1,369 @@
+    });
+
+    it("creates single color attachment", function () {
+      fbm = new FramebufferManager();
+      fbm.update(context, 1, 1);
+      var texture = fbm.getColorTexture();
+      expect(texture).toBeDefined();
+      expect(texture.width).toEqual(1);
+      expect(texture.height).toEqual(1);
+      expect(texture.pixelFormat).toEqual(PixelFormat.RGBA);
+      expect(texture.pixelDatatype).toEqual(PixelDatatype.UNSIGNED_BYTE);
+    });
+
+    it("creates multiple color attachments", function () {
+      var length = 2;
+      fbm = new FramebufferManager({
+        colorAttachmentsLength: length,
+      });
+      fbm.update(context, 1, 1);
+      for (var i = 0; i < length; ++i) {
+        var texture = fbm.getColorTexture(i);
+        expect(texture).toBeDefined();
+        expect(texture.width).toEqual(1);
+        expect(texture.height).toEqual(1);
+        expect(texture.pixelFormat).toEqual(PixelFormat.RGBA);
+        expect(texture.pixelDatatype).toEqual(PixelDatatype.UNSIGNED_BYTE);
+      }
+    });
+
+    it("creates depth attachments", function () {
+      fbm = new FramebufferManager({
+        color: false,
+        depth: true,
+        supportsDepthTexture: true,
+      });
+      fbm.update(context, 1, 1);
+      if (context.depthTexture) {
+        var texture = fbm.getDepthTexture();
+        expect(texture).toBeDefined();
+        expect(texture.width).toEqual(1);
+        expect(texture.height).toEqual(1);
+        expect(texture.pixelFormat).toEqual(PixelFormat.DEPTH_COMPONENT);
+        expect(texture.pixelDatatype).toEqual(PixelDatatype.UNSIGNED_INT);
+      } else {
+        var renderbuffer = fbm.getDepthRenderbuffer();
+        expect(renderbuffer).toBeDefined();
+        expect(renderbuffer.width).toEqual(1);
+        expect(renderbuffer.height).toEqual(1);
+        expect(renderbuffer.format).toEqual(
+          RenderbufferFormat.DEPTH_COMPONENT16
+        );
+      }
+    });
+
+    it("creates depth-stencil attachments", function () {
+      fbm = new FramebufferManager({
+        color: false,
+        depthStencil: true,
+        supportsDepthTexture: true,
+      });
+      fbm.update(context, 1, 1);
+      if (context.depthTexture) {
+        var texture = fbm.getDepthStencilTexture();
+        expect(texture).toBeDefined();
+        expect(texture.width).toEqual(1);
+        expect(texture.height).toEqual(1);
+        expect(texture.pixelFormat).toEqual(PixelFormat.DEPTH_STENCIL);
+        expect(texture.pixelDatatype).toEqual(PixelDatatype.UNSIGNED_INT_24_8);
+      } else {
+        var renderbuffer = fbm.getDepthStencilRenderbuffer();
+        expect(renderbuffer).toBeDefined();
+        expect(renderbuffer.width).toEqual(1);
+        expect(renderbuffer.height).toEqual(1);
+        expect(renderbuffer.format).toEqual(RenderbufferFormat.DEPTH_STENCIL);
+      }
+    });
+
+    it("creates renderbuffer depth attachments if supportsDepthTexture is false", function () {
```

Comment:
This behavior may cause errors to be avoided where previously they would be thrown. i.e., if a class used to always create depth textures without checking `context.depthTexture`, an error would be thrown when the extension was unsupported. Instead, this now creates a renderbuffer if `supportsDepthTexture` is disabled or if the extension is unsupported. 

I noticed this in `PickDepthFramebuffer` and `PointCloudEyeDomeLighting`, but there might be a couple similar places.